src/configurationProvider.ts

```diff
@@ -0,0 +1,101 @@
+     * except the first in a workspace. Currently, this only works if the requested folder is the same as the server's solution path or folder.
+     */
+    private checkWorkspaceInformationMatchesWorkspaceFolder(folder: vscode.WorkspaceFolder | undefined): boolean {
+        const solutionPathOrFolder: string = this.server.getSolutionPathOrFolder();
+
+        // Make sure folder, folder.uri, and solutionPathOrFolder are defined.
+        if (!folder || !folder.uri || !solutionPathOrFolder)
+        {
+            return false;
+        }
+
+        let serverFolder = solutionPathOrFolder;
+        // If its a .sln file, get the folder of the solution.
+        if (fs.lstatSync(solutionPathOrFolder).isFile())
+        {
+            serverFolder = path.dirname(solutionPathOrFolder);
+        }
+
+        // Get absolute paths of current folder and server folder.
+        const currentFolder = path.resolve(folder.uri.fsPath);
+        serverFolder = path.resolve(serverFolder);
+
+        return currentFolder && folder.uri && isSubfolderOf(serverFolder, currentFolder);
+    }
+
+    /**
+	 * Returns a list of initial debug configurations based on contextual information, e.g. package.json or folder.
+	 */
+    provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration[]> {
+        return serverUtils.requestWorkspaceInformation(this.server).then(info => {
+            const generator = new AssetGenerator(info);
+            if (this.checkWorkspaceInformationMatchesWorkspaceFolder(folder) && containsDotNetCoreProjects(info)){
+                const dotVscodeFolder: string = path.join(folder.uri.fsPath, '.vscode');
+                const tasksJsonPath: string = path.join(dotVscodeFolder, 'tasks.json');
+                
+                // Make sure .vscode folder exists, addTasksJsonIfNecessary will fail to create tasks.json if the folder does not exist. 
+                fs.ensureDirSync(dotVscodeFolder);
+    
+                // if the file does not exist, addTasksJson
+                const addTasksJson: boolean = !fs.existsSync(tasksJsonPath);
+    
+                return addTasksJsonIfNecessary(generator, {addTasksJson: addTasksJson}).then(() => {
+                    const isWebProject = generator.hasWebServerDependency();
+                    const launchJson: string = generator.createLaunchJson(isWebProject);
+    
+                    // jsonc-parser's parse function parses a JSON string with comments into a JSON object. However, this removes the comments. 
+                    return parse(launchJson);
+                });
+            }
+            else {
+                // Error to write default C# configurations.
+                throw new Error("Does not contain dotnet core projects.");
```

Comment:
If this message gets surfaced to the user, that should be ".NET Core", not "dotnet core"

Reply:
It does not get surfaced to the user since its caught by the .catch(), but I will still change it.